app/uk/gov/hmrc/securemessage/services/SecureMessageService.scala

```diff
@@ -165,16 +199,77 @@ class SecureMessageService @Inject()(
   private def getIdentifiersFromParticipants(participants: Participants): NonEmptyList[Identifier] =
     participants.participants.map(p1 => p1.identifier)
 
-  private def getParticipantId(client: String, conversationId: String, enrolments: Enrolments)(
-    implicit ec: ExecutionContext): Future[Option[Int]] =
+  private def getParticipant(client: String, conversationId: String, enrolments: Enrolments)(
+    implicit ec: ExecutionContext): Future[Option[Participant]] =
     repo.getConversationParticipants(client, conversationId).map {
       case Some(participants) =>
         val filteredParticipants = enrolmentMaybeParticpant(enrolments, participants).map(identifier =>
           participants.participants.filter(p => p.identifier.equals(identifier)))
-        val firstMatching = filteredParticipants.flatMap(x => x.headOption)
-        firstMatching.map(f => f.id)
+        filteredParticipants.flatMap(x => x.headOption)
+      case _ => None
+    }
+
+  private def getCaseworkerParticipant(client: String, conversationId: String, caseworkerIdentifier: SystemIdentifier)(
+    implicit ec: ExecutionContext): Future[Option[Participant]] = {
+    val identifier = Identifier(caseworkerIdentifier.name, caseworkerIdentifier.value, None)
+    repo.getConversationParticipants(client, conversationId).map {
+      case Some(participants) =>
+        participants.participants.filter(p => p.identifier.equals(identifier)).headOption.map { participant =>
+          participant
+        }
       case _ => None
     }
+  }
+
+  private def messageContentCheck(content: String): Future[Either[SecureMessageError, Boolean]] =
+    if (Base64.isBase64(content)) {
```

Comment:
refactor ifs to functions returning Future[Either[SecureMessageError, Boolea]]
and this can be two lines of for comprehension on the above.
Multilevel if/else is discouraged as it is hard to understand.